tests/unit_tests/test_validators.py

```diff
@@ -161,3 +166,52 @@ def test_summary_validators(mocker):
 [1] ./tests/unit_tests/test_assets/article1.txt
 [2] ./tests/unit_tests/test_assets/article2.txt"""
     )
+
+
+@register_validator("mycustomhellovalidator", data_type="string")
+def validate(value: Any, metadata: Dict[str, Any]) -> ValidationResult:
+    if "hello" in value.lower():
+        return FailResult(
+            error_message="Hello is too basic, try something more creative.",
+            fix_value="hullo",
+        )
+    return PassResult()
+
+
+def test_custom_func_validator():
```

Comment:
Can you add a test here to make sure that functional validators can also be passed in through pydantic?

E.g.
```
class MyModel(BaseModel):
    a: str = Field(..., description='...', validators=[TwoWords(...), validate])
```

Here, I'm not sure if we'd want to add the validator as a reference to the function name (`validate`) or by its string name (`'mycustomhellovalidator'`). Let me know if you have ideas.